ambry-router/src/test/java/com.github.ambry.router/NonBlockingRouterTest.java

```diff
@@ -289,6 +294,71 @@ public void testRequestResponseHandlerThreadExitFlow()
   }
 
   /**
+   * Test that if a composite blob put fails, the successfully put data chunks are deleted.
```

Comment:
The need to clean up only arises in the case of composite blobs where the overall operation can fail after successfully putting individual chunk blobs.